plugin/pkg/admission/podtolerationrestriction/admission.go

```diff
@@ -108,7 +108,16 @@ func (p *Plugin) Admit(ctx context.Context, a admission.Attributes, o admission.
 	}
 	// Final merge of tolerations irrespective of pod type.
 	if len(extraTolerations) > 0 {
-		pod.Spec.Tolerations = tolerations.MergeTolerations(pod.Spec.Tolerations, extraTolerations)
+		mergedOldTolerations := tolerations.MergeTolerations(pod.Spec.Tolerations, nil)
```

Comment:
I haven't studied the logic carefully, please correct me if I'm wrong. Intuitively, this call looks a bit strange. Can we modify the logic instead of calling the same method repeatedly with different parameters? This may make it difficult to understand the code.